src/test/java/core/basesyntax/RegistrationServiceImplTests.java

```diff
@@ -0,0 +1,142 @@
+    }
+
+    @Test
+    void register_userPasswordEqualsNull_Ok() {
+        user.setPassword(null);
+        Assertions.assertThrows(RegistrationServiceException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userPasswordMinLength_Ok() {
+        User actual = storageDao.add(user);
+        Assertions.assertEquals(user, actual);
+    }
+
+    @Test
+    void register_userPasswordEqualsMinLength_Ok() {
+        String passwordMinLength = "a".repeat(PASSWORD_MIN_LENGTH);
+        user.setPassword(passwordMinLength);
+        User actual = storageDao.add(user);
+        Assertions.assertEquals(user, actual);
+    }
+
+    @Test
+    void register_userPasswordMinLength_notOk() {
+        user.setPassword("12345");
+        Assertions.assertThrows(RegistrationServiceException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userAgeMin_Ok() throws RegistrationServiceException {
+        User actual = storageDao.add(user);
+        Assertions.assertEquals(user, actual);
+    }
+
+    @Test
+    void register_userAgeEqualsMin_Ok() throws RegistrationServiceException {
+        user.setAge(AGE_MIN);
+        User actual = storageDao.add(user);
+        Assertions.assertEquals(user, actual);
+    }
+
+    @Test
+    void register_userAgeEqualsMax_Ok() throws RegistrationServiceException {
+        user.setAge(AGE_MAX);
+        User actual = storageDao.add(user);
+        Assertions.assertEquals(user, actual);
+    }
+
+    @Test
+    void register_userAgeMin_notOk() throws RegistrationServiceException {
+        user.setAge(16);
+        Assertions.assertThrows(RegistrationServiceException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userNegativeAgeMin_Ok() throws RegistrationServiceException {
+        user.setAge(-16);
+        Assertions.assertThrows(RegistrationServiceException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userAgeEqualsZero_Ok() throws RegistrationServiceException {
+        user.setAge(0);
+        Assertions.assertThrows(RegistrationServiceException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userAgeEqualsNull_Ok() throws RegistrationServiceException {
+        user.setAge(null);
+        Assertions.assertThrows(RegistrationServiceException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userAgeMoreThanMax_notOk() throws RegistrationServiceException {
+        user.setAge(AGE_MAX + 1);
+        Assertions.assertThrows(RegistrationServiceException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userNull_Ok() {
```

Comment:
1. not OK as we expect exception
2. why we expect NPE and not our RegistrationException?